From e0d093ab8b5a10cc9872cfdc56fe26eacb170cb9 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Wed, 7 Oct 2020 11:40:36 -0500
Subject: [PATCH 63/71] arm64: dts: beacon imx8mm/n kits: Fix PMIC IRQ

The IRQ is currently being configured as GPIO which is wrong,
the pinctrl-name is missing, so the pinmux doesn't happen correctly,
and the pull-up is missing on the IRQ pin.

This patch fixes all three issues on both Mini and Nano dev kits
from Beacon Emebedded Works.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi | 5 +++--
 arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
index eee3a89a4072..0948c4c020f4 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
@@ -65,9 +65,10 @@
 	pmic@4b {
 		reg = <0x4b>;
 		compatible = "rohm,bd71847";
+		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_pmic>;
 		interrupt-parent = <&gpio1>;
-		interrupts = <3 GPIO_ACTIVE_LOW>;
+		interrupts = <3 IRQ_TYPE_LEVEL_LOW>;
 		rohm,reset-snvs-powered;
 
 		regulators {
@@ -345,7 +346,7 @@
 
 		pinctrl_pmic: pmicirq {
 			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x41
+				MX8MM_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x141
 			>;
 		};
 
diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi
index 30d378f86053..fb20facb5c62 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi
@@ -65,9 +65,10 @@
 	pmic@4b {
 		reg = <0x4b>;
 		compatible = "rohm,bd71847";
+		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_pmic>;
 		interrupt-parent = <&gpio1>;
-		interrupts = <3 GPIO_ACTIVE_LOW>;
+		interrupts = <3 IRQ_TYPE_LEVEL_LOW>;
 		rohm,reset-snvs-powered;
 
 		regulators {
@@ -323,7 +324,7 @@
 
 	pinctrl_pmic: pmicirq {
 		fsl,pins = <
-			MX8MN_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x41
+			MX8MN_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x141
 		>;
 	};
 
-- 
2.17.1

